source/Ship.cpp

```diff
@@ -4177,7 +4177,11 @@ void Ship::DoGeneration()
 		if(currentSystem)
 		{
 			double scale = .2 + 1.8 / (.001 * position.Length() + 1);
-			fuel += currentSystem->RamscoopFuel(attributes.Get("ramscoop"), scale);
+			fuel += currentSystem->RamscoopFuel(attributes.Get("ramscoop"), 1., scale);
+			fuel += currentSystem->RamscoopFuel(attributes.Get("velocity ramscoop"),
+				velocity.Length() * !IsHyperspacing() / 1.66, scale);
+			heat += scale * currentSystem->SolarWind() * attributes.Get("ramscoop heat");
+			energy -= scale * currentSystem->SolarWind() * attributes.Get("ramscoop energy");
```

Comment:
I think this should scale with the amount of fuel collected, as that includes velocity-based ramscoop effects properly.

Reply:
I agree that heat and energy should not scale with `ramscoop` or `velocity ramscoop` since that would make those stats multipliers as well, just as `solar heat` is only affected by star `power` and not also `solar collection`. However, a `velocity ramscoop heat` and `velocity ramscoop energy` stat that is affected by both star `wind` and current velocity would make sense.